File: test_projects/openai_gpt_4o_mini_java/src/main/java/com/example/demo/UserService.java
package com.example.demo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

@Service
public class UserService {

    private List<User> users = new ArrayList<>();
    private long nextId = 1;

    public User createUser(User user) {
        user.setId(nextId++);
        users.add(user);
        return user;
    }

    public List<User> getAllUsers() {
        return users;
    }

    public User getUserById(long id) {
        return users.stream().filter(user -> user.getId() == id).findFirst().orElse(null);
    }

    public boolean updateUser(long id, User user) {
        Optional<User> existingUserOpt = users.stream().filter(u -> u.getId() == id).findFirst();
        if (existingUserOpt.isPresent()) {
            User existingUser = existingUserOpt.get();
            existingUser.setName(user.getName());
            existingUser.setEmail(user.getEmail());
            return true;
        }
        return false;
    }

    public boolean deleteUser(long id) {
        return users.removeIf(user -> user.getId() == id);
    }
}

File: test_projects/meta_llama_llama_3_3_70b_instruct_kotlin/src/main/kotlin/com/complang/api/services/UserService.kt

package com.complang.api.services

import com.complang.api.models.User
import com.complang.api.repositories.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class UserService {
    @Autowired
    private lateinit var userRepository: UserRepository

    fun createUser(user: User): User {
        return userRepository.save(user)
    }

    fun getUsers(): List<User> {
        return userRepository.findAll()
    }

    fun getUser(id: Int): User? {
        return userRepository.findById(id).orElse(null)
    }

    fun updateUser(id: Int, user: User): User? {
        val existingUser = getUser(id)
        if (existingUser != null) {
            existingUser.name = user.name
            existingUser.email = user.email
            return userRepository.save(existingUser)
        }
        return null
    }

    fun deleteUser(id: Int) {
        userRepository.deleteById(id)
    }
}
